### PR TITLE
Update prices to 1.0.0 beta 3

### PR DESCRIPTION
**Pull request by rafalp**
This is *work in progress* PR that updates our lib to beta 3, thus making it work with latest changes in our `prices` library.

**TODO**

* [x] Update Model fields
* [x] Update Form fields
* [x] Update template filters
* [x] Make tests pass
* [x] Add validators
* [x] Decide on `decimal_places` handling
* [x] Finalize docs

### CODE REVIEW

**Comment by patrys** [comment]
Could we provide any useful validators? Some examples:

* a validator that checks if the entered value can be expressed within the decimal places allowed by the currency
* a validator that sets min amount
* a validator that sets max amount

**Comment by patrys** [comment]
Let's remove Django 2.0 from allowed failures, it's a stable release now.

**Comment by rafalp** [comment]
I've added `MinMoneyValidator`, `MaxMoneyValidator` and `MoneyValidator` that check if money is at least, at most or precise enough. Each of those validators is just layer of abstraction above the Django's validator: `MinValueValidator`, `MaxValueValidator` and the `DecimalValidator`

File: tests/test_prices.py [hunk 1]
```diff
@@ -170,14 +190,39 @@ def test_field_passes_none_validation():
...
 
+def test_validate_max_money():
+    validator = MaxMoneyValidator(Money(5, 'BTC'))
+    validator(Money('5.00', 'BTC'))
+    with pytest.raises(ValidationError):
+        validator(Money('5.01', 'BTC'))
+
+
+def test_validate_min_money():
+    validator = MaxMoneyValidator(Money(5, 'BTC'))
```
Comment (elwoodxblues) [inline tests/test_prices.py]: Shouldn't it be `MinMoneyValidator`?

**Comment by rafalp** [comment]
Following explicit better than implicit rule I've changed the `MoneyPrecisionValidator` to rely on passed `decimal_places` instead of resolving that on its own via asking babel how many `decimal_places` to use. This validator now closely follows [DecimalValidator](https://docs.djangoproject.com/en/2.0/ref/validators/#decimalvalidator) that it extends, except it works with Money and currencies instead of Decimals.

**Comment by patrys** [comment]
Doesn't that mean that the common use case will require the user to call `babel` manually?

I think a typical case would be to check that a value fits the minimum of:
* what was passed as `decimal_places` (likely coming from the model and reflecting of what the underlying DB field can store)
* what is the correct decimal setting of the currency

**Comment by rafalp** [comment]
> Doesn't that mean that the common use case will require the user to call `babel` manually?

Yeah. I've landed there after looking under the hood of Django's `DecimalField` handling. Here, they have nice Model -> Field -> Validator, passing decimal_places at all times. In my code it was Model -> Field, and then validator just disregards what's passed from Field it and goes for its own from currency. This felt like breaking the pattern and I've redid it to play by same rules everywhere.

> I think a typical case would be to check that a value fits the minimum of:

This is good idea. Picking from there I'm wondering if there would be any downsides if we've simply made`decimal_places` optional, only required when dealing with custom currency that `babel` fails for, eg. like dogecoins, otherwise being resolved automatically from your currency's name. This could be done in our model and form fields as well as in validator

**Comment by patrys** [comment]
I think the problem with `decimal_places` being optional is that it will be provided by Django by default. If that means ignoring the currency precision then I'm afraid in most common cases the currency will be ignored.

**Comment by rafalp** [comment]
All-right. Validator raises in either of those two are true:

- `decimal_places` is specified and shorter than fractional, this is opt-in because `decimal_places` is optional arg in `DecimalValidator` that we are using behind the courtains.
- `currency` is recognized by `babel.numbers.is_currency` and fractional is longer than `get_currency_precision(currency)`, so writing `MoneyPrecisionValidator('USD', 9, 3)` by mistake will still keep you safe because because thanks to babel we'll also test against `decimal_places=2`. Finally `MoneyPrecisionValidator('BTC', 15, 10)` also works if somebody has usecase for such, we just don't know what precision BTC allows and will roll with whatever was configured.

**Comment by rafalp** [comment]
Another pass: changed validation on model field to pass `Money` to validators instead of `Money.amount`  and also create `MoneyPrecisionValidator` for itself just like how `DecimalField` creates `DecimalValidator` for itself. I've also iterated on our tests suite a little.

**Pull request merged**